File: src/service/MoneyService.java
package service;

import model.Money;

import java.util.Scanner;

public class MoneyService {
    public static void printMoney(Money money) {
        System.out.println("Money: " + "\n  sold: " + money.getSold());
    }

    public static void deposit(Money money, Scanner sc) {
        System.out.println("Money to deposit: ");
        if (sc.hasNextDouble()) {
            money.setSold(money.getSold() + sc.nextDouble());
            System.out.println("Deposit successfully");
        } else {
            sc.next();
            System.out.println("Invalid Input - retry");
        }
    }

    public static void withdrawal (Money money, Scanner sc) {
        System.out.println("Amount to withdraw: ");
        while (!sc.hasNextInt()) {
            System.out.println("Invalid Input");
            sc.next();
        }
        double amount = sc.nextInt();
        if (amount > money.getSold()) {
            System.out.println("Insufficient balance");
        }
        else {
            money.setSold(money.getSold() - amount);
            System.out.println("withdrawal successfully");
        }
    }
}

File: src/service/money.service.ts
import { Money } from "../model/money.model";
import { myPrompt } from "../util/my.prompt.util";

export const printMoney = (money: Money): void => {
  console.log(`Money: \n - sold: ${money.getSold()}`);
}

export const deposit = async (money: Money) => {
  const input = await myPrompt("Money to deposit: ");
  if (!isNaN(Number(input))) {
    money.setSold(money.getSold() + Number(input));
    return Promise.resolve("Deposit successfully");
  } else {
    return Promise.reject(new Error("Invalid Input - retry"));
  }
}

export const withdrawal = async (money: Money) => {
  const input = await myPrompt("Amount to withdraw: ");
  if (!isNaN(Number(input))) {
    const amount = Number(input);
    if (amount > money.getSold()) {
      return Promise.reject(new Error("Insufficient balance"));
    } else {
      money.setSold(money.getSold() - amount);
      return Promise.resolve("Withdrawal successfully");
    }
  } else {
    return Promise.reject(new Error("Invalid Input - retry"));
  }
}